src/software/thunderscope/play/playinfo_widget.py

```diff
@@ -0,0 +1,89 @@
+import pyqtgraph as pg
+import pyqtgraph.console as pg_console
+from proto.play_info_msg_pb2 import PlayInfo
+from software.networking.threaded_unix_listener import ThreadedUnixListener
+import software.thunderscope.constants as constants
+from google.protobuf.json_format import MessageToDict
+from pyqtgraph.Qt import QtCore, QtGui
+from pyqtgraph.Qt.QtWidgets import *
+import queue
+
+from proto.robot_log_msg_pb2 import RobotLog
+
+
+class playInfoWidget(QTableWidget):
+
+    NUM_ROWS = 6
+    NUM_COLS = 4
+
+    def __init__(self, buffer_size=10):
+
+        QTableWidget.__init__(self, playInfoWidget.NUM_ROWS, playInfoWidget.NUM_COLS)
+
+        self.log_buffer = queue.Queue(buffer_size)
+        self.verticalHeader().setVisible(False)
+
+        self.setStyleSheet(
+            "QHeaderView::section {background-color: rgba(0, 0, 0, 255); color: rgba(255, 255, 255, 255);}"
+        )
+
+    def set_data(self, data):
+        """Data to set in the table
+
+        :param data: dict containing {"column_name": [column_items]}
+
+        """
+        horizontal_headers = []
+
+        for n, key in enumerate(sorted(data.keys())):
+            horizontal_headers.append(key)
+
+            for m, item in enumerate(data[key]):
+                newitem = QTableWidgetItem(item)
+                self.setItem(m, n, newitem)
+                self.item(m, n).setBackground(QtGui.QColor(0, 0, 0))
+                self.item(m, n).setForeground(QtGui.QBrush(QtGui.QColor(255, 255, 255)))
+
+        self.setHorizontalHeaderLabels(horizontal_headers)
+
+        for x in range(1, playInfoWidget.NUM_ROWS):
+            newitem = QTableWidgetItem(item)
+            self.setItem(x, 0, newitem)
+            self.item(x, 0).setBackground(QtGui.QColor(0, 0, 0))
```

Comment:
i was testing somemore, i think this block is causing a bug where `item` is being cached from the previous loop and displaying incorrectly, you can remove this

Reply:
Those spaces in the table will have a white background, is that fine?

Reply:
yup, ill immediately fix the theme in my next pr